### ISSUE TITLE
[Bug]: Johto Berry Master Trade Bug

### ISSUE THREAD

**Issue by DeeterTrain** [pokeclicker/pokeclicker#5141]
### Confirmations

- [X] I have searched open and closed issues for any existing report of my issue

### What happened?

Johto Berry Master trade purchase of 100 Dusk Balls. Berry inventory contained enough for x33.
Button froze after purchase, locking the option to change the value of purchase quantity any further.
![Pokeclicker Berry Trade Bug](https://github.com/pokeclicker/pokeclicker/assets/161913700/97db1474-3e85-4656-8332-10384f00cb99)
Reopening the trade window or changing game areas did not fix this.
Restarting the desktop client returned the value to the default purchase value of 1 but allowed no further interaction. Denial of further action potentially due to insufficient trade currency.
![Pokeclicker Berry Trade Bug2](https://github.com/pokeclicker/pokeclicker/assets/161913700/5fef2f9c-52bb-4e7e-95dd-d33cb83cb324)
Save file attached, after bug occurred, prior to restarting client.
[[v0.10.19] PokeClicker 2024-03-02 04_34_42.txt](https://github.com/pokeclicker/pokeclicker/files/14468100/v0.10.19.PokeClicker.2024-03-02.04_34_42.txt)

### What did you expect to happen?

Normal button behaviour, the ability to change the purchase value back to the default or to view another option.

### Steps to Reproduce

1. Visit Johto Berry Master.
2. Make berry trade purchase in excess of current inventory amount.
3. Bug is encountered, unable to alter trade value.

### What operating system are you using?

Windows 10

### What browsers are you seeing the problem on?

Desktop Client

**Comment by AevitasDragonkin** [comment]
Yes, it's due to not a single trade being available. That being said, it's reworked and fixed with #5124
